feat: setup react page with submenu for datasources listview

nytai opened the pull request:
### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
- Sets up infra for rendering a react view under `databaseview/list/` when `ENABLE_REACT_CRUD_VIEWS` config flag is enabled and `SIP_34_DATABASE_UI` feature flag is enabled. 
- adds a submenu to the database list page. 
- moves props shared between datasets and databases subMenu into a common file.

- renames the "welcome" app to "crud_views" and moves the welcome view under `views/CRUD/` 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
<img width="1679" alt="Screen Shot 2020-08-19 at 11 21 03 AM" src="https://user-images.githubusercontent.com/10255196/90674558-236d3300-e20e-11ea-8b14-584b9b6a8ba2.png">

### TEST PLAN
<!--- What steps should be taken to verify the changes -->
- FAB view renders when `ENABLE_REACT_CRUD_VIEWS` and `SIP_34_DATASOURCE_UI` are both off.
- React view (only showing SubMenu) renders when `ENABLE_REACT_CRUD_VIEWS` and `SIP_34_DATASOURCE_UI` are both on.

- unit tests for DatasourceList
- CI
### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

nytai committed:
feat: set up react page with submenu for datasources view

nytai committed:
add submenu for datsources

nytai committed:
black

kalimuthu123 reviewed (commented):
good processs

riahk commented on superset-frontend/src/views/CRUD/dataset/DatasetList.tsx:
```diff
@@ -448,7 +448,11 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
       },
-      { name: 'Databases', label: t('Databases'), url: '/databaseview/list/' },
+      {
+        name: 'Dasources',
```
```suggestion
        name: 'Datasources',
```

nytai replied:
thanks for catching that!

riahk reviewed (commented):
It's probably not in scope (especially with how routing is currently setup), but we should be aiming to get all the lists within data to share a `SubMenu` to avoid repeating it in each list view.

nytai committed:
Update superset-frontend/src/views/CRUD/dataset/DatasetList.tsx

Co-authored-by: Moriah Kreeger <moar.riah@gmail.com>

nytai commented:
@riahk I agree, this copying of the subMenu around is less than ideal. You very much identified the challenge, the way routing is set up. Ideally routing would look something like `/data/datasources` and `/data/datasets` and the SubMenu would live under the `/data` route portion. What's preventing that change is that we're currently feature flagging these views and have to support legacy routes. Once we fully deprecate the legacy views (and routes, potentially) we can set up permanent redirects.

mistercrunch commented:
There are really "Databases". The term "datasource" ambiguated with "dataset". Let's use "Databases". Maybe we need a short glossary in the docs.

mistercrunch commented on superset-frontend/src/views/CRUD/dataset/DatasetList.tsx:
```diff
@@ -448,7 +448,11 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
+      {
+        name: 'Datasources',
+        label: t('Datasources'),
+        url: '/databaseview/list/',
```
I vote for moving to `/database/list/`. The "view" suffix is a FAB artifact from what the ModelView class was class.

nytai committed:
Datasource -> Database

nytai committed:
move shared menu props into a common file

nytai committed:
move welcome page under views/CRUD rename welcome app to crud_views

nytai committed:
SIP_34_DATASOURCE_UI -> SIP_34_DATABASE_UI

nytai commented:
@riahk took a shot at sharing a menu component between Datasets and Databases, but quickly realized that only half the menu is shared. There's logic that conditionally adds the "Bulk Select" and "+Dataset" actions, those are specific to the Datasets view. Instead I just moved the props that are shared into a common file. 

I also renamed the "Welcome" app to something more semantic and moved the Welcome component under `src/views/CRUD`

rusackas commented on superset-frontend/src/components/Modal.tsx:
```diff
@@ -20,7 +20,7 @@ import React from 'react';
 import { Modal as BaseModal } from 'react-bootstrap';
 import { t } from '@superset-ui/translation';
-import Button from 'src/views/CRUD/dataset/Button';
+import Button from 'src/views/CRUD/data/dataset/Button';
```
FWIW, I'm planning to open a PR soon sweeping through and cleaning out any one-off buttons, including this fella. One button to rule them all! ;) I know this is the newest-looking button, so I'll be supporting various bits and pieces from it.

nytai replied:
Yes, the new buttons are this one and the really wide CTA one. Maybe they can just be variants of the `bsSize` attribute

rusackas replied:
right... maybe a "cta" prop... to apply these styles would do just fine. It just needs a good name, really :D Anyway, we can discuss this outside of this PR.

rusackas commented on superset-frontend/src/views/App.tsx:
```diff
@@ -30,14 +30,15 @@ import Menu from 'src/components/Menu/Menu';
+import setupApp from 'src/setup/setupApp';
+import setupPlugins from 'src/setup/setupPlugins';
+import Welcome from 'src/views/CRUD/welcome/Welcome';
+import ToastPresenter from 'src/messageToasts/containers/ToastPresenter';
```
thanks for catching these... they're _everywhere_

rusackas commented on superset-frontend/src/views/CRUD/welcome/Welcome.tsx:
```diff
@@ -21,8 +21,8 @@ import { Panel, Row, Col, Tabs, Tab, FormControl } from 'react-bootstrap';
 import { User } from 'src/types/bootstrapTypes';
-import RecentActivity from '../profile/components/RecentActivity';
-import Favorites from '../profile/components/Favorites';
+import RecentActivity from '../../../profile/components/RecentActivity';
```
```suggestion
import RecentActivity from 'src/profile/components/RecentActivity';
```

rusackas commented on superset-frontend/src/views/CRUD/welcome/Welcome.tsx:
```diff
@@ -21,8 +21,8 @@ import { Panel, Row, Col, Tabs, Tab, FormControl } from 'react-bootstrap';
-import RecentActivity from '../profile/components/RecentActivity';
-import Favorites from '../profile/components/Favorites';
+import RecentActivity from '../../../profile/components/RecentActivity';
+import Favorites from '../../../profile/components/Favorites';
```
```suggestion
import Favorites from 'src/profile/components/Favorites';
```

rusackas commented on superset-frontend/webpack.config.js:
```diff
@@ -180,7 +180,7 @@ const config = {
     dashboard: addPreamble('/src/dashboard/index.jsx'),
     sqllab: addPreamble('/src/SqlLab/index.tsx'),
-    welcome: addPreamble('/src/welcome/index.tsx'),
+    crud_views: addPreamble('/src/views/index.tsx'),
```
any reason not to follow the camelCase pattern here?

nytai replied:
Sine these are consumed on the python side, I thought snake case made sense. Realizing now that there's another bundle that's camelCased

nytai replied:
switched to camelCase for consistency

nytai committed:
use absolute imports; camelcase bundle name

rusackas reviewed (approved):
Looks like there are a few nits questions remaining that sound like they're being addressed, but I don't see any deal-breakers, so LGTM!
